### PR TITLE
WFS requests using POST

### LINKED ISSUE #297: Test OWSLib PR allowing POST for WFS `getfeature` requests
## Description

We need to make WFS requests for info on many subregions. For a large number of regions, the GET requests exceeds the maximum size. See https://github.com/geopython/OWSLib/issues/439

Francis made a PR https://github.com/geopython/OWSLib/pull/706 to solve this, and we'd need a review of the PR, as well as confirmation that it solves our issue, illustrated in https://pavics-sdi.readthedocs.io/projects/raven/en/latest/notebooks/Subset_climate_data_over_watershed.html#Subsetting-a-gridded-climate-dataset
https://github.com/Ouranosinc/raven/blob/master/docs/source/notebooks/gridded_data_subset.ipynb

The notebook sets the number of features to 2, but it should work with 100s of features. 

TODO: 
- [x] Local install of PR branch
- [x] Add `method="POST"` to `flyingpigeon.subset_base.get_feature` 
- [ ] Create unit test checking that it works with large number of features (that would crash GET requests).
- [ ] Confirm PR is functional for this test case - report success in OWSLib PR comments
- [ ] Wait for PR to be merged and new version of OWSLib released
- [ ] Make new flyingpigeon release
- [ ] Update flyingpigeon version in raven requirements

### PR DESCRIPTION
**Pull request by Zeitsperre**
## Overview

This PR fixes #297 

Changes:

* Added `POST` support for one of the WFS-enabled WPS requests (`hydrobasins-select`)
* Added `method` flags to specify the methods of requests (`{"GET", "POST"}`) that can be requested.

## Related Issue / Discussion
https://github.com/geopython/OWSLib/pull/706

## Additional Information

Links to other issues or sources.

### CODE REVIEW

**Comment by Zeitsperre** [comment]
@huard this seems to be working using the fixes and workarounds developed by @f-PLT. POST is now implemented for the hydrobasins-select WPS process.

File: raven/utilities/gis.py [hunk 1]
```diff
@@ -380,13 +401,17 @@ def get_hydrobasins_attributes_wfs(
                 outputFormat="json",
                 filter=filterxml,
             )
+            q = Request(method=method, url=GEO_URL, params=params).prepare().url
 
-            q = Request("GET", GEO_URL, params=params).prepare().url
+            # NOTE: This could be rewritten for OWSLib to return the dataset instead of the URL if the need arises
+            # from owslib.wfs import WebFeatureService
+            # wfs = WebFeatureService(url=GEO_URL, version="1.1.0", timeout=30)
+            # req = wfs.getfeature(**params)
```
Comment (huard) [inline raven/utilities/gis.py:409]: ?

**Review by huard — COMMENTED** [review]
I don't think this function ever had an issue with GET requests exceeding a limit. I realized looking into this the culprit was in flyingpigeon. So not clear this PR is necessary.

File: raven/utilities/gis.py (hunk 1 above)
Comment (Zeitsperre) [reply raven/utilities/gis.py:409]: Oh, this was just a thought. I can remove this.

**Comment by Zeitsperre** [comment]
I leave the decision of merging to you.

**Pull request closed without merging**